### PR TITLE
Update BAP to build for new ISO API

### PR DESCRIPTION
**Pull request by Thalley**
Update the BAP implementation to build and run on the new ISO API.

### CODE REVIEW

**Comment by LingaoM** [comment]
May be we need add “Bluetooth Audio” label ?

**Comment by Thalley** [comment]
> May be we need add “Bluetooth Audio” label ?

We actually have that label, but it is only applied for the `subsys/bluetooth/audio` (where the upstreamed LE audio stuff is placed), and this is placed in `subsys/bluetooth/host/audio`. I guess we could/should add temporary codeowners and label commits for this branch. @asbjornsabo thoughts?

File: include/bluetooth/audio.h [hunk 1]
```diff
@@ -36,13 +36,13 @@ struct bt_audio_cap;
 
 #define BT_BROADCAST_ID_SIZE             3 /* octets */
 
-#if defined(CONFIG_BT_BAP)
+#if defined(CONFIG_BT_BAP) && defined(CONFIG_BT_AUDIO_BROADCAST)
```
Comment (asbjornsabo) [inline include/bluetooth/audio.h:39]: I guess we eventually want to BAP and BROADCAST to depend on each other.  (One will depend upon the other, not sure now which one will be highest in the hierarchy.)
(That is outside the scope of this PR.)
When that is done, it will be sufficient to check for one of these.

(Ditto for endpoint.h)

File: subsys/bluetooth/host/audio/bap.c [hunk 2]
```diff
@@ -138,8 +138,13 @@ static int bap_reconfig(struct bt_audio_chan *chan,
 		tmp->codec = codec;
 	}
 
-	/* Unbind if there is an existing QoS */
-	bt_audio_chan_unbind(chan);
+	/* Terminate CIG if there is an existing QoS,
```
Comment (asbjornsabo) [inline subsys/bluetooth/host/audio/bap.c:141]: I may need some education here: The following code seems to terminate the CIG unconditionally.  Where is the "if there is an existing QoS"?

File: subsys/bluetooth/host/audio/chan.c [hunk 3]
```diff
@@ -70,6 +71,7 @@ struct bt_audio_base_ad {
 	struct bt_audio_base_ad_subgroup subgroups[BROADCAST_SUBGROUP_CNT];
 } __packed;
 
+static struct bt_iso_cig *cigs[BROADCAST_SRC_CNT];
```
Comment (asbjornsabo) [inline subsys/bluetooth/host/audio/chan.c:74]: Should this be BT_BAP_CONNECTED_AUDIO_GROUP_COUNT instead?

File: subsys/bluetooth/host/audio/chan.c [hunk 4]
```diff
@@ -1071,51 +1067,92 @@ struct bt_conn_iso *bt_audio_chan_bind(struct bt_audio_chan *chan,
...
+
+		err = bt_iso_cig_create(&param, free_cig);
+		if (err != 0) {
+			BT_ERR("bt_iso_cig_create failed: %d", err);
 			return NULL;
 		}
 	}
 
-	return &chan->iso->conn->iso;
+	return &chan->iso->iso->iso;
```
Comment (asbjornsabo) [inline subsys/bluetooth/host/audio/chan.c:1103]: I still find these triplicates confusing ...

File: tests/bluetooth/audio_shell/prj.conf [hunk 5]
```diff
@@ -17,9 +17,8 @@ CONFIG_BT_MAX_CONN=4
 
 CONFIG_BT_AUDIO=y
 CONFIG_BT_AUDIO_UNICAST=y
-CONFIG_BT_AUDIO_BROADCAST=n
-CONFIG_BT_BAP=n
-CONFIG_BT_ISO_BROADCAST=n
+CONFIG_BT_AUDIO_BROADCAST=y
+CONFIG_BT_BAP=y
```
Comment (asbjornsabo) [inline tests/bluetooth/audio_shell/prj.conf:21]: Also re-enable BASS and BASS_CLIENT (with debugging)?

**Review by asbjornsabo — CHANGES_REQUESTED** [review]

File: include/bluetooth/audio.h (hunk 1 above)
Comment (Thalley) [reply include/bluetooth/audio.h:39]: That is true. Currently they are separate entities. 

I think that in time we will get rid of the CONFIG_BT_BAP and simply use `CONFIG_BT_AUDIO_{UNICAST/BROADCAST}`.

File: subsys/bluetooth/host/audio/bap.c (hunk 2 above)
Comment (Thalley) [reply subsys/bluetooth/host/audio/bap.c:141]: Added a check

File: subsys/bluetooth/host/audio/chan.c (hunk 4 above)
Comment (Thalley) [reply subsys/bluetooth/host/audio/chan.c:1103]: Agreed. 

I'd rather not change the chan->iso in this PR as this is just fixing stuff. 

The other `iso->iso` could probably also be improved upon in some way, but what is upstream.

**Comment by asbjornsabo** [comment]
> > May be we need add “Bluetooth Audio” label ?
> 
> We actually have that label, but it is only applied for the `subsys/bluetooth/audio` (where the upstreamed LE audio stuff is placed), and this is placed in `subsys/bluetooth/host/audio`. I guess we could/should add temporary codeowners and label commits for this branch. @asbjornsabo thoughts?

We have removed the CODEOWNERS for this branch, to avoid noise.  I think that still applies (especially as we will still do rebases).
I usually add labels like the "audio" one by hand, it is not much of a hassle.

File: tests/bluetooth/audio_shell/prj.conf [hunk 6]
```diff
@@ -83,7 +82,7 @@ CONFIG_BT_AUDIO_DEBUG_CHAN=y
 CONFIG_BT_AUDIO_DEBUG_ENDPOINT=y
 CONFIG_BT_AUDIO_DEBUG_CAPABILITIES=y
 CONFIG_BT_DEBUG_ASCS=y
-CONFIG_BT_DEBUG_BAP=n
+CONFIG_BT_DEBUG_BAP=y
 CONFIG_BT_DEBUG_BASS=n
 CONFIG_BT_DEBUG_BASS_CLIENT=n
```
Comment (asbjornsabo) [inline tests/bluetooth/audio_shell/prj.conf:87]: Also these two debug ones, please?

File: tests/bluetooth/audio_shell/prj.conf (hunk 5 above)
Comment (Thalley) [reply tests/bluetooth/audio_shell/prj.conf:21]: Fixed

**Review by asbjornsabo — APPROVED** [review]
LGTM
I would like to have debug for BASS and BASS_CLIENT enabled again as well, as that is how it used to be.  (But it is not critical at all.)

File: tests/bluetooth/audio_shell/prj.conf (hunk 6 above)
Comment (Thalley) [reply tests/bluetooth/audio_shell/prj.conf]: Done
